genesis/src/lib.rs

```diff
@@ -65,12 +65,16 @@ impl GenesisNetwork {
                 .clone()
                 .ok_or_else(|| eyre!("Genesis account private key is empty."))?,
         )?;
-        let transactions_iter = raw_block.transactions.into_iter();
         #[cfg(not(test))]
-        let transactions_iter =
-            transactions_iter.chain(std::iter::once(GenesisTransactionBuilder {
-                isi: vec![UpgradeBox::new(Validator::try_from(raw_block.validator)?).into()],
-            }));
+        // First instruction should be Validator upgrade.
+        // This makes possible to grant permissions to users in genesis.
```

Comment:
Excellent. This needs to be documented in the official docs.

Reply:
This does not affect user api, because validator is specified in a special field and we automatically move it to be the first instruction